src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/mailer.html.twig

```diff
@@ -139,28 +139,31 @@
                                                     </div>
                                                 </div>
                                             </div>
-                                            {% if message.htmlBody is defined %}
+                                            {% if message.htmlBody() is defined %}
                                                 {# Email instance #}
-                                                <div class="tab">
-                                                    <h3 class="tab-title">HTML preview</h3>
-                                                    <div class="tab-content">
-                                                        <pre class="prewrap" style="max-height: 600px">
-                                                            <iframe
-                                                                src="data:text/html;base64;charset=utf-8,{{ collector.base64Encode(message.htmlBody()) }}"
-                                                                style="height: 80vh;width: 100%;"
-                                                            >
-                                                            </iframe>
-                                                        </pre>
+                                                {% set htmlBody = message.htmlBody() %}
+                                                {% if htmlBody is not null %}
+                                                    <div class="tab">
+                                                        <h3 class="tab-title">HTML Preview</h3>
+                                                        <div class="tab-content">
+                                                            <pre class="prewrap" style="max-height: 600px">
+                                                                <iframe
+                                                                    src="data:text/html;charset=utf-8;base64,{{ collector.base64Encode(htmlBody) }}"
```

Comment:
This fixes https://github.com/symfony/symfony/issues/44696

Reply:
`htmlBody` could also be of type `resource` and will create a `TypeError`.

Reply:
do you know how can I test this locally? providing a `resource` to test/fix it if needed

Reply:
I've seen in the doc:

`->html(fopen('/path/to/emails/user_signup.html', 'r'))` I will test this in demo app to see how it goes

Reply:
It looks like the HTML Content tab has the same problem with `resource` because Twig filter `convert_encoding` uses `iconv()` internally.

Reply:
> will create a TypeError

it did not when using example above